src/averageAges.js

```diff
@@ -55,7 +66,35 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const filteredChildren = (!onlyWithSon)
+    ? peopleFilter(people, person => people.find(
+      mother => mother.name === person.mother
+    ))
+    : peopleFilter(people, person => people.find(
+      mother => mother.name === person.mother && person.sex === 'm'
+    ));
```

Comment:
DRY, select needed callback inside `peopleFilter` instead of writing function twice